source/Octopus.Tentacle/Kubernetes/KubernetesJobService.cs

```diff
@@ -63,6 +66,39 @@ await foreach (var (type, job) in response.WatchAsync<V1Job, V1JobList>(onError,
             }
         }
 
+        public async Task<V1JobList> ListAllJobsAsync(CancellationToken cancellationToken)
+        {
+            return await Client.ListNamespacedJobAsync(KubernetesConfig.Namespace,
+                labelSelector: OctopusLabels.ScriptTicketId,
+                cancellationToken: cancellationToken);
+        }
+
+        public async Task WatchAllJobsAsync(string initialResourceVersion, Func<WatchEventType, V1Job, Task> onChange, Action<Exception> onError, CancellationToken cancellationToken)
+        {
+            using var response = Client.BatchV1.ListNamespacedJobWithHttpMessagesAsync(
+                KubernetesConfig.Namespace,
+                labelSelector: OctopusLabels.ScriptTicketId,
+                resourceVersion: initialResourceVersion,
+                watch: true,
+                cancellationToken: cancellationToken);
+
+            var watchErrorCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+
+            Action<Exception> internalOnError = ex =>
+            {
+                //We cancel the watch explicitly (so it can be restarted)
+                watchErrorCancellationTokenSource.Cancel();
+
+                //notify there was an error
+                onError(ex);
+            };
```

Comment:
My read of the source code for `WatchAsync` is that it won't kill the watch onError, so this allows us to cancel the `WatchAsync` if there is an error, so we can then restart it.